Android/unityLibrary/hybidwrapper.androidlib/src/main/java/net/pubnative/unityplugin/HyBidAdViewWrapper.java

```diff
@@ -0,0 +1,130 @@
+package net.pubnative.unityplugin;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.util.Log;
+
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+
+import com.unity3d.player.UnityPlayer;
+
+import net.pubnative.lite.sdk.HyBid;
+import net.pubnative.lite.sdk.models.AdSize;
+import net.pubnative.lite.sdk.views.HyBidAdView;
+
+public class HyBidAdViewWrapper implements HyBidAdView.Listener {
+
+    private static final String TAG = HyBidAdViewWrapper.class.getSimpleName();
+
+    protected HyBidAdView hyBidAdView;
+
+    protected String mGameObjectName;
+    protected String mAdId;
+
+    protected void executeDisplayAction(Runnable runnable) {
+        if (UnityPlayer.currentActivity == null) {
+            Log.e(TAG, "No active context found to show the ad");
+        } else {
+            UnityPlayer.currentActivity.runOnUiThread(runnable);
+        }
+    }
+
+    protected void setGameObject(String gameObjectName) {
+        this.mGameObjectName = gameObjectName;
+    }
+
+    protected void setAdId(String adId) {
+        this.mAdId = adId;
+    }
+
+    public void load(String gameObjectName, String appToken, final String placementId, String adId, int position) {
+        if (ContextCompat.checkSelfPermission(UnityPlayer.currentActivity.getApplicationContext(),
+                Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+            doHyBidStuff(gameObjectName, appToken, placementId, adId, position);
+        } else {
+            ActivityCompat.requestPermissions(UnityPlayer.currentActivity,
+                    new String[]{Manifest.permission.ACCESS_FINE_LOCATION},
+                    10);
+        }
+    }
+
+    private void doHyBidStuff(String gameObjectName, final String appToken, final String placementId, String adId, final int position) {
+
+        try {
+            setGameObject(gameObjectName);
+            setAdId(adId);
+
+            if (hyBidAdView != null) {
+                UnityPlayer.currentActivity.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        hyBidAdView.destroy();
+                        loadBanner(appToken, placementId, position);
+                    }
+                });
+            } else {
+                loadBanner(appToken, placementId, position);
+            }
+        } catch (Exception e) {
+            Log.d("Exception", e.toString());
+        }
+    }
+
+    private void loadBanner(String appToken, final String placementId, int position) {
+        final HyBidAdView.Position bannerPosition;
+
+        if (position == getTopPosition()) {
```

Comment:
can be replaced with ternary operator to shorten

Reply:
There is only 1 method loadBanner and it is called in all cases

Reply:
Cool ,will do this